lib/srv/desktop/rdp/rdpclient/client.go

```diff
@@ -435,23 +449,25 @@ func (c *Client) handleRemoteCopy(data []byte) C.CGOErrCode {
 }
 
 // Wait blocks until the client disconnects and runs the cleanup.
-func (c *Client) Wait() error {
+func (c *Client) Wait() {
 	c.wg.Wait()
-	// Let the Rust side free its data.
-	C.free_rdp(c.rustClient)
-	return nil
+	c.Cleanup()
 }
 
-// Close shuts down the client and closes any existing connections.
-// It is safe to call multiple times, from multiple goroutines.
-// Calls other than the first one are no-ops.
-func (c *Client) Close() {
-	c.closeOnce.Do(func() {
+// Cleanup frees the memory of the cgo.Handle, closes the RDP client connection,
+// and frees the Rust client.
+func (c *Client) Cleanup() {
+	c.cleanupOnce.Do(func() {
+		// Release the memory of the cgo.Handle
 		c.handle.Delete()
```

Comment:
Shouldn't we move `handle.Delete()` after `C.close_rdp()` to ensure that the Rust code  is done running and won't try to use the handle after we've deleted it?